### DISCUSSION TITLE
Stuck on enabling hidden 4g decoding

### DISCUSSION THREAD

**Discussion by ArchOcean1** [xCuri0/ReBarUEFI#183] [Q&A]
i don't understand what i should do after i downloaded IRFExtractor can you help?

**Comment by MrAnonymous7122** [comment]
The guide on enabling hidden 4g decoding was recently updated.

**Comment by janstadt** [comment]
Did you get any further? Im also kinda stuck on that 4g decoding step as well. I have no knowledge of how to do this stuff, but i postd on win-raid some info: https://winraid.level1techs.com/t/dell-precision-5820/101490 basically looking to enable rebar on an older dell precision 5820. I included the lates BIOS update file and was able (i think) to extract bin files, but i end up with a bunch of different bin files. Any help would be greatly appreciated.

**Comment by MrAnonymous7122** [comment]
Which step of the guide are you still on?

**Reply by janstadt** [reply]
Step 5 here: https://github.com/xCuri0/ReBarUEFI/wiki/Enabling-hidden-4G-decoding it doesnt output a txt file. I am assuming its due to the fact that i have a dell bios update file and when i extracted the bin from it (i cant even remember how i did that, i think its in the link above) its not the correct file or something?

**Reply by MrAnonymous7122** [reply]
So you have ran irfextractor on the correct .bin, what file did it produce and did it give any errors? Post console log here

**Reply by janstadt** [reply]
Im not sure if i did it on the correct bin or not i guess. I ran this: https://github.com/platomav/BIOSUtilities?tab=readme-ov-file#dell-pfs-update-extractor on this https://dl.dell.com/FOLDER11491866M/1/M33X_2.36.0.exe and it produced 2 folders and each had 5 or 6 bin files. I chose the one with the word BIOS in it and run UFEITool on it and find the `4G Decod` and extract that body (some files had a lot of hits with that string as well which i was unsure about). I then used the irfextractor on the body file and the output is nothing. No logs, nothing in the console and it produces nothing. This is all foreign to me so i appreciate the patience and questions. I really dont know what im doing but would love to get rebar enabled on this machine.

**Reply by MrAnonymous7122** [reply]
Afaik, this doesn't work on OEM builds but we can try get it to work. Can you show what uefitool shows when you search for 4g decod?

**Reply by MrAnonymous7122** [reply]
Could you also try using winrar to extract the .exe and see if there is a bios file hidden within?

**Reply by janstadt** [reply]
I initially tried that and it seems as though there is a file in there that might be a bin but it has no extension and is named [0]. I will see if i can get some more info after work here.

**Reply by janstadt** [reply]
Here's what the exe looks like after opening it in 7z.
<img width="1420" alt="Screenshot 2024-05-15 at 9 31 07 PM" src="https://github.com/xCuri0/ReBarUEFI/assets/1872087/7f959eba-6f88-490a-8bc4-2efa86fe646e">

**Reply by MrAnonymous7122** [reply]
You could try extract the .exe and see if there is a bios file there

**Reply by MrAnonymous7122** [reply]
The m33x file

**Reply by janstadt** [reply]
Sorry, i should have been more clear. Thats the bio exe that i downloaded from dell. The other files are what is extracted.

**Reply by janstadt** [reply]
@MrAnonymous7122 Would you be able to download that exe file i included and see if you can poke around to determine if its even possible to do all this? Would really love to get this all figured out.

**Reply by janstadt** [reply]
Also, i see a name `@zhiwoo` listed in the known working flashes but i cant find them anywhere. I see they flashed a Dell 5810 and i have a 5820 so i am hoping its something that is achievable.

**Reply by MrAnonymous7122** [reply]
I'm not really an expert on OEM bios files, it's possible the updater only contains code to download the bios directly from dell. Seeing as it is an OEM, it's worth looking at a physical hardware flasher which can read and write directly to and from the bios chip. From there you can edit the current bios and then reflash the updated one.

**Reply by GezelligLLC** [reply]
we own a dell 7920 and a 5820. We'd like to modify the bios too. Did you have any success with your dells?
